### PR TITLE
[2017-10][interp] fixes for interp AOT mode

### PR DESCRIPTION
**Pull request by lewurm**
backport of https://github.com/mono/mono/pull/6112

### CODE REVIEW

File: mono/mini/aot-compiler.c [hunk 1]
```diff
@@ -11389,6 +11396,35 @@ add_preinit_got_slots (MonoAotCompile *acfg)
 	acfg->nshared_got_entries = acfg->got_offset;
 }
 
+static const char* interp_in_static_sigs[] = {
```
Comment (kumpera) [inline mono/mini/aot-compiler.c]: How are those signatures computed?

Comment (lewurm) [reply mono/mini/aot-compiler.c]: see commit message https://github.com/mono/mono/pull/6122/commits/b64faae88c30da83c842fc89edb8eb9f2a6174d9

Comment (kumpera) [reply mono/mini/aot-compiler.c]: Why can't this be computed by the AOT compiler?

Comment (lewurm) [reply mono/mini/aot-compiler.c]: These wrappers are needed for native2managed transitions. We would need to know all callsites from the embedder that are using the `compile_method ()` callback of the runtime and determine the used signatures. That's not possible with the AOT compiler.

Comment (kumpera) [reply mono/mini/aot-compiler.c]: We should then require all embeders to annotate all managed functions they plan to call with a cattr we can look for during AOT. Similar to how the reverse pinvoke one works.

Comment (vargaz) [reply mono/mini/aot-compiler.c]: For bitcode, we simply generate wrappers for all signatures in the app. With sharing, its not that big.

Comment (lewurm) [reply mono/mini/aot-compiler.c]: @kumpera that would still require us to scan all assemblies ahead-of-time which we can't always do

Comment (kumpera) [reply mono/mini/aot-compiler.c]: Absolutely but this setup is really fragile.
BCL + SDKs should not be manually curated.

Comment (lewurm) [reply mono/mini/aot-compiler.c]: We are thinking about getting rid of this requirement to know ahead of time all the signatures, and do something similar for what we do for the `managed2native` transition today (aka. the `enter_icall_trampoline`.

**Pull request merged**